src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -417,6 +417,14 @@ def delete_role_assignments(cmd, ids=None, assignee=None, role=None,
         for i in ids:
             assignments_client.delete_by_id(i)
         return
+    if not any([ids, assignee, role, resource_group_name, scope, assignee, yes]):
+        from knack.prompting import prompt_y_n, NoTTYException
+        try:
+            msg = 'You are deleting all role assignments under the subscription. Are you sure to continue?'
```

Comment:
Knack already handles the "NoTTYException" so you shouldn't have to. On wording, I would recommend: "This will delete all role assignments under the subscription. Are you sure?"